.github/workflows/i18n-consistency-checker.yaml

```diff
@@ -17,88 +12,70 @@ jobs:
-          \\n\
-          The following files may have consistency issues with the English version. Please check and update the files.\\n\
-          \\n\
-          This issue is created when there is an update to content/en. It compares the git update history to let you know what updates are overdue. The issue should be closed when the update is complete.\\n"
-          
-          for file in $(find patterns/2-structured -name '*.md'); do
-          
-            # Get the translated file name and check if it exists
-            i18n_filename=$(echo "$file" | sed 's/patterns\/2-structured/translation\/${{matrix.language}}\/patterns/g')
+          # Declare the flags
+          declare -A flags=( ["ja"]=":jp: Japanese" ["zh"]=":cn: Chinese")
+        
+          issue_title="${flags['${{matrix.language}}']}: Content Consistency Issue"
+        
+          # Heredoc for issue header
+          cat <<- EOM > issue.md
+          # i18n Contents Consistency Issue
+        
+          The following files may have consistency issues with the English version. Please check and update the files.
+        
+          This issue is created when there is an update to content/en. It compares the git update history to let you know what updates are overdue. The issue should be closed when the update is complete.
+          EOM
+        
+          # Loop through all files in the English directory
+          for file in $(find patterns/{2-structured,3-validated} -name '*.md'); do
+            [[ $file =~ "3-validated" ]] && continue # if the file is under 3-validated, skip (one liner) - 2023/08/26
+            i18n_filename=$(echo "$file" | sed "s/patterns\/\(2-structured\|3-validated\)/translation\/${{matrix.language}}\/patterns/g")
+            
             if [[ ! -e "$i18n_filename" ]]; then
-            	continue
+              continue
             fi
-            echo $file
-            echo $i18n_filename
-            # Loop through all the files in the English directory
-            # Get the last updated date of the original file and the translated file
-            original_updated_at=$(date -d "$(git log -1 --format=%cd --date=iso $file)" +%s) 
+        
+            original_updated_at=$(date -d "$(git log -1 --format=%cd --date=iso $file)" +%s)
             i18n_content_updated_at=$(date -d "$(git log -1 --format=%cd --date=iso $i18n_filename)" +%s)
-             
-            # print the last updated date of the original file and the translated file
-            
-            # Check if the translated file is updated after the original file
-            if [[ $(($original_updated_at - $i18n_content_updated_at)) -ge 1 ]]; then
-              # Get the title of the content
-              content_header=$(echo "$(cat "$file")" | grep -E '^title+' | sort -r | head -n1)
+        
+            if [[ $((original_updated_at - i18n_content_updated_at)) -ge 1 ]]; then
+              content_header=$(grep -E '^title+' "$file" | sort -r | head -n1)
```

Comment:
Doesn't look like the title extract works right now

However let's leave this in here, as we may be adding a frontmatter to all patterns anyways, at which point this should work :)

Reply:
No wonder this thing doesn't work.
I was getting the title from the "title: foobar" meta info to match hugo's md style... But this time it's gitbook, so I need to extract from ## Title.
I didn't realize this when I ported...

I've changed it to use awk to extract it, so it should work now!

<img width="963" alt="image" src="https://github.com/InnerSourceCommons/InnerSourcePatterns/assets/15963767/d61a894c-070e-4706-8897-513e1d7b84a2">

Reply:
Very pretty!